### ISSUE TITLE
Cannot read property 'html' of undefined

### ISSUE THREAD

**Issue by rochapablo** [FortAwesome/angular-fontawesome#11]
I'm getting the follow error:

```
ERROR TypeError: Cannot read property 'html' of undefined
at FaIconComponent.ngOnChanges (angular-fontawesome.es5.js:106)

// @TODO: make sure that it doesn't break things to do html[0] here.
this.renderedIconHTML = this.sanitizer.bypassSecurityTrustHtml(renderedIcon.html[0]);
```

**app.module**
```
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';

imports: [
    ...
    FontAwesomeModule
  ],
```

**component.html**
```
<fa-icon [icon]="faUser"></fa-icon>
```

**package.json**
```
"@angular": "^5.2.0",
"@fortawesome/angular-fontawesome": "0.1.0-3",
"@fortawesome/fontawesome": "^1.1.3",
"@fortawesome/fontawesome-free-solid": "^5.0.6",
```

What could I be missing?

**Comment by mlwilkerson** [comment]
Thanks for jumping in here, @rochapablo.

Looks like in your `component.html`, you are trying to reference the `faUser` _object_, but have you made that object available to your template?

If you look in the `examples` of this repo, you'll see how the [ExampleComponent](https://github.com/FortAwesome/angular-fontawesome/blob/master/examples/example.component.ts) makes a few icon objects available to the template.

Basically, your options are:
1. Reference the icon object, in which case, that icon object needs to be made available to the template. So you'd then do `[icon]="faUser"` in the template. OR,
1. Reference the icon name as a string, in which, you would have to have first added the icon object to the icon library (see the coffee icon in the `ExampleComponent`). So you'd then do `icon='coffee'` in the template.

Does that help?

**Comment by mlwilkerson** [comment]
What must be happening is that the call to the `icon()` API function is returning `undefined`, indicating that it didn't find an icon, given `iconSpec`. And `iconSpec` is derived from  `[icon]="faUser"`.

So, regardless, I think we should have better error handling here.

**Comment by rochapablo** [comment]
Yeh, you're right! I got this working.

Now I'm trying to just use [string and/or array](https://github.com/FortAwesome/angular-fontawesome/blob/bb7c3a1dbae48339095f11028d82889609a08972/src/icon.component.ts#L29) to get something like ```fas fa-address-book``` but still getting undefined.

Also is it possible don't have to load all icons that I will be using?

**Comment by rochapablo** [comment]
Well I believe that this will make a huge bundle, but it will include every possible icon

```
import solid from '@fortawesome/fontawesome-free-solid';
import regular from '@fortawesome/fontawesome-free-regular';
import brands from '@fortawesome/fontawesome-free-brands';
import fontawesome from '@fortawesome/fontawesome';

fontawesome.library.add(solid);
fontawesome.library.add(regular);
fontawesome.library.add(brands);
```

**Comment by mlwilkerson** [comment]
I'm working on adding something to the example in the repo that will clarify how to reference icons with non-default prefixes.

As for your workaround, while that may work, it is essentially avoiding the Angular component and instead relying on our `autoReplaceSvg` functionality to translate `<i>` tags into `<svg>`. If you're satisfied with that, great. But you may find it more performant (including having smaller bundle size) to import only those icons that you intend to use and add them to the library, or make them available as objects to your templates.

I'd like to leave this issue open for the moment, because I do want to fix the error handling problem you stumbled onto here.

**Comment by mlwilkerson** [comment]
OK, I've just published a new pre-release `0.1.0-4` which has some expanded examples and more explanation both in the source code under `examples/` and in the example app web output. So if you were to `yarn start` (or `npm run start`) on this latest prerelease, and look at both the browser output and the `examples/` source code, I think it may help.

This is also relevant to the discussion on #10 because it demonstrates using the array syntax for specifying an icon's prefix, as would be necessary to specify a style prefix other than the `fas` default.  For example, this shows how you'd get that `far` prefix on the `flag` icon. The example code shows two ways to do it:

1. Making the `faFlag` icon _object_ from the `fontawesome-free-regular` icon pack available to the template, like this:
```
<fa-icon [icon]="faFlag"></fa-icon>
```
OR

2. Adding that `faFlag` icon object to the library and then referencing it by `iconName` as a string—along with the `far` style prefix as a string—from your template, like this:
```
<fa-icon [icon]="['far','user']"></fa-icon>
```

**Comment by rochapablo** [comment]
Thank you!

```
<fa-icon [icon]="['far','user']"></fa-icon>
```

Worked just as I expected.

Well done sir.

**Comment by rochapablo** [comment]
Just an observation; using in a "normal" component like the [example](https://github.com/FortAwesome/angular-fontawesome/tree/master/examples) it works normally. But when I use inside an reusable component it will keeps getting the ```Cannot read property 'html' of undefined```.

Here's an example

```
@Component({
  selector: 'app-icon',
  template: `
    <fa-icon [icon]="icon"></fa-icon> <!-- Cannot read property 'html'... -->
    <!-- <i [ngClass]="icon"></i> no problem here -->
  `
})
export class AppIconComponent implements OnChanges, OnInit {
  @Input() public icon: Array<string>;
   ...
}
```

**Comment by robmadole** [comment]
That looks like a separate issue, @rochapablo are you willing to file a separate bug for this?

**Comment by mlwilkerson** [comment]
@rochapablo it looks like in your most recent example you haven't set up the icon correctly. But I would have expected the new error logging in version `0.1.0-5` to have given you a more helpful error message that might help with debugging. 

If you haven't already, could you please update `@fortawesome/ember-fontawesome` to that latest prerelease version and try again?

I think the error message will show that we can't find an icon for `prefix: far` and `iconName: fa-flag`. That would be the case for two reasons:
1. the iconName for the `faFlag` object is `flag`, not `fa-flag`. One way to view the iconName is to look at it's entry in the icon package. In this case, look at  `node_modules/@fortawesome/fontawesome-free-regular/faFlag.js`
1. you haven't (or at least I don't see in your code snippet above) added the `faFlag` icon to the library. So if even you used the correct `iconName`, it would not be able to find it.

**Comment by mlwilkerson** [comment]
Whoops, I misread your code snippet. 

I think the part of my previous response about updating to the latest prerelease is still valid and may help with debugging. But I'll take another look.

**Comment by mlwilkerson** [comment]
Oh, so yes, my comment about the `iconName` being `flag` not `fa-flag` is also still valid. It may work correctly if you just change it to:
```
<app-icon [icon]="['far', 'flag']"></app-icon>
```

I think the reason why you're seeing that `<i [ngClass]="icon"></i>` works is that using an `<i>` tag when FontAwesome configured with `autoReplaceSvg: true` causes our library to automagically transform that `<i>` into an `<svg>`. It's bypassing the Angular component bits altogether at that point. And the syntax in that case involves using a _class name_ `fa-flag`, which ends up being the `${FontAwesomeConfig.familyPrefix}-${iconName}` (semi-pseudo-code, there).

But when you're specifying an Array _object_ to the `icon` attribute of the Angular component, it is expecting and array of [`prefix`, `iconName`].

So to summarize, `fa-flag` is not the `iconName`; `flag` is the `iconName`.
`fa-flag` is a _class name_ that is used on an `<i>` tag in the `autoReplaceSvg` usage scenario.

**Comment by rochapablo** [comment]
Yep, ```['far', 'flag']``` without ```fa-``` worked.

Oh about the version, I was using ```@fortawesome/angular-fontawesome```: ```0.1.0-4``` that's why I wasn't getting the right message error, updating to ```0.1.0-5``` worked.

I believe that now we're done here. Thank you again for your time.

**Comment by robmadole** [comment]
@mlwilkerson you da man 💯

**Comment by dhniels** [comment]
@rochapablo could you show me how you got this working in your Angular project? I'm getting this error `Can't bind to 'icon' since it isn't a known property of 'fa-icon'.`

This is how i have it setup so far:

**app.module**

```
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import fontawesome from '@fortawesome/fontawesome'
import solid from '@fortawesome/fontawesome-free-solid';
import regular from '@fortawesome/fontawesome-free-regular';

imports: [
    ...
    FontAwesomeModule
  ],
```

**component.html**

`<fa-icon [icon]="faStickyNote"></fa-icon>`

**package.json**

"@angular": "5",
"@fortawesome/angular-fontawesome": "*",
"@fortawesome/fontawesome": "1",
"@fortawesome/fontawesome-common-types": "*",
"@fortawesome/fontawesome-free-regular": "5",
"@fortawesome/fontawesome-free-solid": "5",
